service/kernel/env_detect.cpp

```diff
@@ -137,18 +137,22 @@ DAAL_EXPORT void daal::services::Environment::initNumberOfThreads()
 {
     if (isInit) return;
 
-    /* if HT enabled - set _numThreads to physical cores num */
+    /* if HT enabled - set number of threads to physical cores num */
     if (daal::internal::Service<>::serv_get_ht())
     {
         /* Number of cores = number of cpu packages * number of cores per cpu package */
-        int ncores = daal::internal::Service<>::serv_get_ncpus() * daal::internal::Service<>::serv_get_ncorespercpu();
+        const size_t ncores = daal::internal::Service<>::serv_get_ncpus() * daal::internal::Service<>::serv_get_ncorespercpu();
 
-        /*  Re-set number of threads if ncores is valid and different to _numThreads */
-        if ((ncores > 0) && (ncores < _daal_threader_get_max_threads()))
+        /*  Re-set number of threads if ncores is valid */
+        if ((ncores > 0) && (ncores <= _daal_threader_get_max_threads()))
```

Comment:
Your modification looks wrong. What are reasons to set it, if ncores == _daal_threader_get_max_threads() ?

Reply:
Original logic was restored